### PR TITLE
Add creator staking endpoints

### PR DESCRIPTION
**Pull request by samchuk-vlad**

### CODE REVIEW

File: src/connections/networks/index.ts
Comment (teodorus-nathaniel) [inline src/connections/networks/index.ts:8]: this one should be changed back to subsocial before merge right?

File: src/constant/index.ts
Comment (teodorus-nathaniel) [inline src/constant/index.ts:19]: change this back before merge

File: src/server.ts
Comment (teodorus-nathaniel) [inline src/server.ts:45]: this one also before merge

File: src/services/creatorStaking/creatorsList.ts
Comment (teodorus-nathaniel) [inline src/services/creatorStaking/creatorsList.ts:32]: from what I know, isn't there toPrimitive() which I think is better for this? so you don't need to remove the , manually

File: src/services/creatorStaking/creatorsSpaces.ts
Comment (teodorus-nathaniel) [inline src/services/creatorStaking/creatorsSpaces.ts:30]: need to update it to subsocial before merge?
but isn't it better to have use env which determines whether to use soonsocial or subsocial?

File: src/services/creatorStaking/creatorsSpaces.ts
Comment (teodorus-nathaniel) [inline src/services/creatorStaking/creatorsSpaces.ts:37]: afaik, for redis, if you set expiry of any item, it will automatically delete the data,
so isn't it possible to just get() and you don't need to check needUpdate() everytime?

File: src/services/creatorStaking/index.ts
Comment (teodorus-nathaniel) [inline src/services/creatorStaking/index.ts:70]: if you just use if (value) here, isn't it same?

**Review by teodorus-nathaniel — APPROVED** [review]

**Review by olehmell — CHANGES_REQUESTED** [review]

File: src/services/rpc.ts
Comment (olehmell) [inline src/services/rpc.ts:1]: We are using the copypaste in 3 different repos, time to extract to library :)

**Review by olehmell — CHANGES_REQUESTED** [review]

File: src/connections/networks/index.ts
Comment (samchuk-vlad) [reply src/connections/networks/index.ts:8]: Yes

File: src/services/creatorStaking/creatorsList.ts
Comment (samchuk-vlad) [reply src/services/creatorStaking/creatorsList.ts:32]: toPrimitive is not working correctly

File: src/services/creatorStaking/creatorsSpaces.ts
Comment (samchuk-vlad) [reply src/services/creatorStaking/creatorsSpaces.ts:37]: > it will automatically delete the data...
 ------------------------------------
 So that why I don't use default redis expiry. I created custom functionality for expire (you can check `src/cache/index.ts` file)

**Pull request merged**
